crates/world/flashblocks/src/payload/job.rs

```diff
@@ -0,0 +1,394 @@
+            BuiltPayload = OpBuiltPayload,
+            Attributes = OpPayloadBuilderAttributes<OpTxEnvelope>,
+        > + Unpin
+        + 'static,
+    Builder::Attributes: Unpin + Clone,
+    Builder::BuiltPayload: Unpin + Clone,
+{
+    type Output = Result<(), PayloadBuilderError>;
+
+    /// Polls the payload builder job to drive payload construction and management.
+    ///
+    /// This implementation follows a state-driven approach with the following logic:
+    ///
+    /// # Flow
+    ///
+    /// 1. **Deadline Check**: First checks if the payload building deadline has been reached.
+    ///    If so, returns [`Poll::Ready(Ok(()))`] to complete the job.
+    ///
+    /// 2. **Interval Tick**: Polls the interval timer to determine when to spawn new build jobs.
+    ///    - If the interval is ready, continues to spawn a new build job
+    ///    - If pending, schedules a wake-up and returns [`Poll::Pending`]
+    ///
+    /// 3. **Pending Block Processing**: If there's a pending block being built:
+    ///    - Processes build outcomes ([`BuildOutcome::Better`], [`BuildOutcome::Freeze`], [`BuildOutcome::Aborted`])
+    ///    - [`BuildOutcome::Better`]:
+    ///         - Each successive payload is a pre-commitment to the next payload.
+    ///         - We will continuously build on top of the best payload until the job is resolved.
+    ///    - [`BuildOutcome::Freeze`]: marks payload as frozen and stops further building
+    ///    - [`BuildOutcome::Aborted`]: handles cancellation and potential respawning
+    ///    - On errors: logs and continues operation
+    ///    - If still pending: restores the future and returns [`Poll::Pending`]
+    ///
+    /// 4. **New Job Spawning**: If no pending block exists, spawns a new build job
+    ///    and continues polling.
+    ///
+    /// # Pre-confirmation Behavior
+    ///
+    /// Each time we hit [`BuildOutcome::Better`], we increment `block_index` and create a new
+    /// pre-confirmation on top of the current best payload. This allows for iterative improvement
+    /// where each successive build acts as a pre-commitment to the next block state.
+    ///
+    /// # Returns
+    ///
+    /// The polling continues until either the deadline is reached or an error occurs, returning
+    /// [`Poll::Pending`] for ongoing work or [`Poll::Ready(Ok(()))`] when complete.
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let _span = span!(target: "payload_builder", tracing::Level::DEBUG, "poll").entered();
+        let _enter = _span.enter();
+        let this = self.get_mut();
+        // check if the deadline is reached
+        if this.deadline.as_mut().poll(cx).is_ready() {
+            trace!(target: "payload_builder", "payload building deadline reached");
+            return Poll::Ready(Ok(()));
+        }
+
+        let mut tick_fut = Box::pin(this.interval.tick());
+
+        while tick_fut.as_mut().poll(cx).is_pending() {
```

Comment:
I'm confused by this, don't we only want to call the waker when tick_fut is Ready?